Mathlib/Data/Nat/Init.lean

```diff
@@ -434,6 +434,45 @@ protected lemma div_left_inj (hda : d ∣ a) (hdb : d ∣ b) : a / d = b / d ↔
   refine ⟨fun h ↦ ?_, congrArg fun b ↦ b / d⟩
   rw [← Nat.mul_div_cancel' hda, ← Nat.mul_div_cancel' hdb, h]
 
+/-- TODO: Replace `Nat.sub_mul_div` in core. -/
+lemma sub_mul_div' (a b c : ℕ) : (a - b * c) / b = a / b - c := by
+  obtain h | h := Nat.le_total (b * c) a
+  · rw [Nat.sub_mul_div _ _ _ h]
+  · rw [Nat.sub_eq_zero_of_le h, Nat.zero_div]
+    by_cases hn : b = 0
+    · simp only [hn, Nat.div_zero, zero_le, Nat.sub_eq_zero_of_le]
+    · have h2 : a / b ≤ (b * c) / b := Nat.div_le_div_right h
+      rw [Nat.mul_div_cancel_left _ (zero_lt_of_ne_zero hn)] at h2
+      rw [Nat.sub_eq_zero_of_le h2]
+
+lemma mul_sub_div_of_dvd (a b c : ℕ) (h1 : c ≠ 0) (h2 : c ∣ b) : (c * a - b) / c = a - b / c := by
+  obtain ⟨_, hx⟩ := h2
+  simp only [hx, ← Nat.mul_sub_left_distrib, Nat.mul_div_right, zero_lt_of_ne_zero h1]
+
+lemma mul_add_mul_div_of_dvd (hb : b ≠ 0) (hd : d ≠ 0) (h1 : b ∣ a) (h2 : d ∣ c) :
+    (a * d + b * c) / (b * d) = a / b + c / d := by
+  obtain ⟨n, hn⟩ := h1
+  obtain ⟨_, hm⟩ := h2
+  rw [hn, hm, Nat.mul_assoc b n d, Nat.mul_comm n d, ← Nat.mul_assoc, ← Nat.mul_assoc,
+    ← Nat.mul_add, Nat.mul_div_right _ (zero_lt_of_ne_zero hb),
+    Nat.mul_div_right _ (zero_lt_of_ne_zero hd),
+    Nat.mul_div_right _ (Nat.mul_pos (zero_lt_of_ne_zero hb) (zero_lt_of_ne_zero hd))]
+
+lemma mul_sub_mul_div_of_dvd (hb : b ≠ 0) (hd : d ≠ 0) (h1 : b ∣ a) (h2 : d ∣ c) :
+    (a * d - b * c) / (b * d)  = a / b - c / d:= by
+  obtain ⟨n, hn⟩ := h1
+  obtain ⟨m, hm⟩ := h2
+  rw [hn, hm]
+  rw [Nat.mul_assoc,Nat.mul_comm n d, ← Nat.mul_assoc,← Nat.mul_assoc, ← Nat.mul_sub_left_distrib,
+    Nat.mul_div_right _ (zero_lt_of_ne_zero hb), Nat.mul_div_right _ (zero_lt_of_ne_zero hd),
+    Nat.mul_div_right _ (Nat.mul_pos (zero_lt_of_ne_zero hb) (zero_lt_of_ne_zero hd))]
+
+lemma div_mul_assoc (a b c : ℕ) (h : b ∣ a) : (a / b) * c = (a * c) / b := by
+  rw [Nat.mul_comm, ← Nat.mul_div_assoc _ h, Nat.mul_comm]
+
+lemma eq_div_iff_mul_eq_left (a b c : ℕ) (h1 : 0 < c) (h2 : c ∣ b) : a = b / c ↔ b = a * c := by
```

Comment:
This is an iff, so give as few explicit arguments as possible 
```suggestion
lemma eq_div_iff_mul_eq_left (hc : 0 < c) (hcb : c ∣ b) : a = b / c ↔ b = a * c := by
```
Same below

Reply:
Done